src/graphql/status.rs

```diff
@@ -52,46 +49,74 @@ struct Properties {
     stats: String,
 }
 
-#[derive(InputObject, SimpleObject, Debug)]
-#[graphql(input_name = "InputPeerList")]
-pub struct PeerList {
-    pub addr: String,
-    pub hostname: String,
-}
+#[Object]
+impl GigantoConfig {
+    async fn ingest_srv_addr(&self) -> String {
+        self.ingest_srv_addr.to_string()
+    }
 
-impl TomlPeers for PeerList {
-    fn get_hostname(&self) -> String {
-        self.hostname.clone()
+    async fn publish_srv_addr(&self) -> String {
+        self.publish_srv_addr.to_string()
     }
-    fn get_addr(&self) -> String {
-        self.addr.clone()
+
+    async fn graphql_srv_addr(&self) -> String {
+        self.graphql_srv_addr.to_string()
     }
-}
 
-#[derive(SimpleObject, Debug)]
-struct GigantoConfig {
-    ingest_srv_addr: String,
-    publish_srv_addr: String,
-    graphql_srv_addr: String,
-    retention: String,
-    max_open_files: i32,
-    max_mb_of_level_base: u64,
-    addr_to_peers: String,
-    peer_list: Vec<PeerList>,
-    ack_transmission_cnt: u16,
+    async fn retention(&self) -> String {
+        humantime::format_duration(self.retention).to_string()
+    }
+
+    async fn data_dir(&self) -> String {
+        self.data_dir.to_string_lossy().to_string()
+    }
+
+    async fn log_dir(&self) -> String {
+        self.log_dir.to_string_lossy().to_string()
+    }
+
+    async fn export_dir(&self) -> String {
+        self.export_dir.to_string_lossy().to_string()
+    }
+
+    async fn max_open_files(&self) -> i32 {
+        self.max_open_files
+    }
+
+    async fn max_mb_of_level_base(&self) -> u64 {
```

Comment:
Why don't you use `StringNumber`?

Reply:
Or, is this Okay it is not directly returned?

Reply:
done